codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Message.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http2;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+
+public abstract class Http2Message {
+    Http2Message() { }
```

Comment:
I think keeping it "closed" until it is more concrete is a fine idea.  However I think we should eventually allow extension.  I'm not sure if this is what you are referring to but I'm no a big fan of the big if/else blocks based upon `instanceof` checks in other codes.  I think I like having a `writer` object or a method directly on the interface to write itself may be more preferable.